src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/ReferenceDataResourceIntTest.kt

```diff
@@ -53,4 +49,46 @@ class ReferenceDataResourceIntTest : IntegrationTestBase() {
         .jsonPath("$[?(@.id == 'NWEB')]").doesNotExist()
     }
   }
+
+  @DisplayName("GET /reference-data/email-domains")
+  @Nested
+  inner class GetActiveEmailDomains {
+    private val spec = webTestClient.get().uri("/reference-data/email-domains")
+
+    @Test
+    fun `access forbidden when no authority`() {
+      spec.exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access allowed even when no roles specified`() {
+      spec.headers(setAuthorisation(roles = listOf()))
+        .exchange()
+        .expectStatus().isOk
+    }
+
+    @Test
+    fun `access allowed for any role specified`() {
+      spec.headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .exchange()
+        .expectStatus().isOk
+    }
+
+    @Test
+    fun `will return all active email domains`() {
+      spec.headers(setAuthorisation())
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("$").value<JSONArray> {
```

Comment:
converting to a `JSONArray` means that we can do standard `assertThat` on the results and when things don't match we get the full contents of the body rather than just being told a single line isn't found etc.